kartothek/core/common_metadata.py

```diff
@@ -78,10 +76,7 @@ def _schema_compat(self):
 
             schema = schema.remove_metadata()
             md = {b"pandas": _dict_to_binary(pandas_metadata)}
-            if ARROW_LARGER_EQ_0150:
-                schema = schema.with_metadata(md)
-            else:
-                schema = schema.add_metadata(md)
+            schema = schema.add_metadata(md)
```

Comment:
```suggestion
            schema = schema.with_metadata(md)
```
causes build to fail